### PR TITLE
JBIDE-24963 - Create CDI 2.0 integration tests - step 1

### PR DESCRIPTION
**Pull request by zcervink**
JBIDE-24963 - Create CDI 2.0 integration tests - step 1
	NamedComponentsSearchingTestCDI20.class
	BeanInjectOpenOnTestCDI20.class
	FindObserverEventTestCDI20.class
	CDIWebProjectWizardTestCDI20.class

Jenkins: https://dev-platform-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/cdi20.itests/4/
Jira: https://issues.jboss.org/browse/JBIDE-24963

Signed-off-by: Zbynek Cervinka <zcervink@redhat.com>

### CODE REVIEW

**Comment by zcervink** [comment]
please review @odockal

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/wizard/template/CDIWebProjectWizardTemplate.java [hunk 1]
```diff
@@ -92,8 +97,13 @@ public void createCDIProject(){
 	
 	//cdi1.1+
 	//cd1.0 should show warning
+	//JBIDE-26643 | cdi2.0 does not contain the page for unchecking the "create beans.xml" option - test skiped
 	@Test
 	public void createCDIProjectWithoutBeansXml(){
+		// JBIDE-26643 | test skipped for CDI 2.0
+		if (CDIVersion.equals("2.0")) {
```
Comment (odockal) [inline tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/wizard/template/CDIWebProjectWizardTemplate.java]: I do not like the way how this is solved, imagine, when the problem is fixed, you have not other way to find out it is fixed unless you watch specific issue and you cannot catch any other possible failure if there is other bug. Rather, I would suggest to create a custom exception that will be thrown at the moment when specific issue occurs. So surround the erroneous code with try catch block, wait for proper exception, and throw your own exception (ie., JBIDEXYZException). In annotation of the test method add a expect param and assign it to your new exception. This design will allow you to have test green, unless there is another bug or issue is fixed.

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/CDI20SuiteTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import org.jboss.tools.cdi.bot.test.beansxml.validation.cdi11.BeansXMLValidationQuickFixTestCDI11;
+import org.jboss.tools.cdi.bot.test.validation.cdi11.CDIValidatorTestCDI11;
+import org.jboss.tools.cdi.bot.test.wizard.cdi11.CDIWebProjectWizardTestCDI11;
+import org.jboss.tools.cdi.bot.test.wizard.cdi20.CDIWebProjectWizardTestCDI20;
+import org.junit.runner.RunWith;
+import org.junit.runners.Suite.SuiteClasses;
+
+@RunWith(RedDeerSuite.class)
+@SuiteClasses({
+//	BeanValidationQuickFixTestCDI11.class,
```
[the remark this answers is not in the document]
Comment (zcervink) [reply tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/CDI20SuiteTest.java]: removed

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beans/openon/cdi20/FindObserverEventTestCDI20.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import org.eclipse.reddeer.requirements.jre.JRERequirement.JRE;
+import org.eclipse.reddeer.requirements.openperspective.OpenPerspectiveRequirement.OpenPerspective;
+import org.eclipse.reddeer.requirements.server.ServerRequirementState;
+import org.jboss.ide.eclipse.as.reddeer.server.family.ServerMatcher;
+import org.jboss.ide.eclipse.as.reddeer.server.requirement.ServerRequirement.JBossServer;
+import org.jboss.tools.cdi.bot.test.beans.openon.template.FindObserverEventTemplate;
+import org.junit.Before;
+
+/**
+ * Test operates on @Named searching
```
[the remark this answers is not in the document]
Comment (zcervink) [reply tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beans/openon/cdi20/FindObserverEventTestCDI20.java]: Almost no other testclass has a javadoc description, so I have left there only @author annotation for now.

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/wizard/template/CDIWebProjectWizardTemplate.java (hunk 1 above)
Comment (zcervink) [reply tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/wizard/template/CDIWebProjectWizardTemplate.java]: Changed as described. Now the test is running as expected for CDI 2.0, but not for CDI 1.0 and CDI 1.2, where no exception will be thrown (and the test expects the exception).

Comment (zcervink) [reply tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/wizard/template/CDIWebProjectWizardTemplate.java]: Fixed - now the test is running for CDI 1.0, 1.2 a 2.0.

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/CDI20SuiteTest.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+package org.jboss.tools.cdi.bot.test;
+
+import org.eclipse.reddeer.junit.runner.RedDeerSuite;
+import org.jboss.tools.cdi.bot.test.beans.named.cdi20.NamedComponentsSearchingTestCDI20;
+import org.jboss.tools.cdi.bot.test.beans.openon.cdi20.BeanInjectOpenOnTestCDI20;
+import org.jboss.tools.cdi.bot.test.beans.openon.cdi20.FindObserverEventTestCDI20;
+import org.jboss.tools.cdi.bot.test.wizard.cdi20.CDIWebProjectWizardTestCDI20;
+import org.junit.runner.RunWith;
+import org.junit.runners.Suite.SuiteClasses;
+
```
Comment (jkopriva) [inline tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/CDI20SuiteTest.java]: missing author

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/wizard/cdi20/CDIWebProjectWizardTestCDI20.java [hunk 5]
```diff
@@ -0,0 +1,114 @@
+/*******************************************************************************
+ * Copyright (c) 2010-2018 Red Hat, Inc.
```
Comment (jkopriva) [inline tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/wizard/cdi20/CDIWebProjectWizardTestCDI20.java]: incorrect copyright year

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beans/openon/cdi20/FindObserverEventTestCDI20.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
+/*******************************************************************************
+ * Copyright (c) 2010-2018 Red Hat, Inc.
```
Comment (jkopriva) [inline tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beans/openon/cdi20/FindObserverEventTestCDI20.java]: incorrect copyright year

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/CDI20SuiteTest.java (hunk 4 above)
Comment (zcervink) [reply tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/CDI20SuiteTest.java]: FIxed.

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/wizard/cdi20/CDIWebProjectWizardTestCDI20.java (hunk 5 above)
Comment (zcervink) [reply tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/wizard/cdi20/CDIWebProjectWizardTestCDI20.java]: FIxed.

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beans/openon/cdi20/FindObserverEventTestCDI20.java (hunk 6 above)
Comment (zcervink) [reply tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beans/openon/cdi20/FindObserverEventTestCDI20.java]: FIxed.

Comment (zcervink) [reply tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beans/openon/cdi20/FindObserverEventTestCDI20.java]: FIxed.

**Comment by psrna** [comment]
Based on the communication with @odockal I'm merging this. Sonar complains will be targeted in a subsequent PR.

**Pull request merged**